src/commands/ingress/tryConfigureIngressUsingDockerfile.ts

```diff
@@ -0,0 +1,105 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { AzExtFsExtra } from '@microsoft/vscode-azext-utils';
+import { ext } from '../../extensionVariables';
+import { localize } from '../../utils/localize';
+import type { IngressContext } from './IngressContext';
+import { getDefaultPort } from './editTargetPort/getDefaultPort';
+
+export async function tryConfigureIngressUsingDockerfile(context: IngressContext): Promise<void> {
+    if (!context.dockerfilePath) {
+        return;
+    }
+
+    context.dockerfileExposePorts = await getDockerfileExposePorts(context.dockerfilePath);
+
+    if (context.alwaysPromptIngress) {
```

Comment:
Small nit, but should probably be at the top of the file so we don't waste any time doing the async call above.

Reply:
This position is intentional, because we want to get the expose ports still so we can provide port suggestions later down the line